src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,296 @@
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleShowCollectionForm = this.handleShowCollectionForm.bind(this);
+		this.handleShowAllCollections = this.handleShowAllCollections.bind(this);
+		this.handleAddToNewCollection = this.handleAddToNewCollection.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	async componentDidMount() {
+		const collections = await this.getCollections();
+		// eslint-disable-next-line react/no-did-mount-set-state
+		this.setState({collectionsAvailable: collections});
+	}
+
+	toggleRow(collectionId) {
+		const oldSelected = this.state.selectedCollections;
+		let newSelected;
+		if (oldSelected.find(selectedId => selectedId === collectionId)) {
+			newSelected = oldSelected.filter(selectedId => selectedId !== collectionId);
+		}
+		else {
+			newSelected = [...oldSelected, collectionId];
+		}
+		this.setState({
+			selectedCollections: newSelected
+		});
+	}
+
+	async getCollections() {
+		// Get all collections of the user (unlikely that a user will have more than 10000 collections
+		const req = await request.get(`/editor/${this.props.userId}/collections/collections?type=${this.props.entityType}&size=10000`);
+		const collections = req.body;
+		return collections;
+	}
+
+	async handleAddToCollection() {
+		const {bbids} = this.props;
+		const {selectedCollections} = this.state;
+		if (selectedCollections.length) {
+			const promiseArray = [];
+			selectedCollections.forEach((collectionId) => {
+				const submissionURL = `/collection/${collectionId}/add`;
+				promiseArray.push(
+					request.post(submissionURL)
+						.send({bbids})
+				);
+			});
+			await Promise.all(promiseArray);
+			this.setState({
+				message: {
+					text: `Successfully added to selected collection${selectedCollections.length > 1 ? 's' : ''}`,
+					type: 'success'
+				}
+			});
+		}
+		else {
+			this.setState({
+				message: {
+					text: 'No Collection Selected',
+					type: 'danger'
+				}
+			});
+		}
+	}
+
+	handleShowCollectionForm() {
+		this.setState({message: {}, showCollectionForm: true});
+	}
+
+	handleShowAllCollections() {
+		this.setState({message: {}, showCollectionForm: false});
+	}
+
+	handleAddToNewCollection(evt) {
+		evt.preventDefault();
+
+		if (!this.isValid()) {
+			this.setState({
+				message: {
+					text: 'Internal Error: Try again later',
+					type: 'danger'
+				}
+			});
+			return;
+		}
+
+		const description = this.description.getValue();
+		const name = this.name.getValue();
+		const privacy = this.privacy.getValue();
+		const {entityType} = this.props;
+
+		const data = {
+			description,
+			entityType,
+			name,
+			privacy
+		};
+		const {bbids} = this.props;
+		request.post('/collection/create/handler')
+			.send(data)
+			.then((res) => {
+				request.post(`/collection/${res.body.id}/add`)
+					.send({bbids}).then(() => {
+						this.setState({
+							message: {
+								text: `Successfully added to your new collection: ${name}`,
+								type: 'success'
+							}
+						});
```

Comment:
This looks nice currently, but a user is allowed to continue clicking on "add to new collection" and it will create as many duplicate collections.
We can avoid that easily by either:
1. closing the modal (and maybe showing a success alert in the entity footer (but what about adding from the search page?) or 
2. disabling the button after successful creation. I think that is more elegant and self-contained.